rest-core/src/Rest/Error.hs

```diff
@@ -46,11 +45,11 @@ domainReason = CustomReason . DomainReason
 infixl 3 >|<
 -- | Combine two ExceptT computations yielding the last error if both fail.
 -- This prevents the need for a Semigroup or Monoid instance for the error type, which is necessary if using (<!>) or (<|>) respectively.
-(>|<) :: (Applicative m, Monad m) => ExceptT e m a -> ExceptT e m a -> ExceptT e m a
-a >|< b = mapE getLast (mapE Last a <!> mapE Last b)
-  where
-    ExceptT m <!> ExceptT n = ExceptT $ do
-      v <- m
-      case v of
-        Left e -> fmap (either (Left . (<>) e) Right) n
-        Right x -> return (Right x)
+(>|<) :: (Applicative m, Monad m) => ExceptT f m a -> ExceptT e m a -> ExceptT e m a
+ExceptT m >|< ExceptT n = ExceptT $ m >>= either (const n) (return . Right)
+
+infixl 3 >?<
+-- | Try to get a Just from Maybe computations (left to right), otherwise gives Nothing
+-- Example: f >?< g `orThrow` NotFound
+(>?<) :: (MonadTrans t, Monad (t m), Monad m) => m (Maybe a) -> m (Maybe a) -> t m (Maybe a)
+(>?<) m n = lift m >>= maybe (lift n) (return . Just)
```

Comment:
So it would be either

```haskell
(>?<) :: (Monad m) => m (Maybe a) -> m (Maybe a) -> m (Maybe a)
(>?<) m n = m >>= maybe n (return . Just)
```

or

``` haskell
(>?<) :: (Functor m, Monad m) => m (Maybe a) -> m (Maybe a) -> m (Maybe a)
(>?<) m n = runMaybeT $ MaybeT m <|> MaybeT n
```